src/Files.App/Utils/Taskbar/SystemTrayIcon.cs

```diff
@@ -0,0 +1,338 @@
+					PInvoke.Shell_NotifyIcon(NOTIFY_ICON_MESSAGE.NIM_DELETE, in lpData);
+
+					_notifyIconCreated = true;
+
+					// Add a new icon
+					PInvoke.Shell_NotifyIcon(NOTIFY_ICON_MESSAGE.NIM_ADD, in lpData);
+
+					lpData.Anonymous.uVersion = 4u;
+
+					// Set the icon handler version
+					// NOTE: Do not omit this code. If you remove, the icon won't be shown.
+					PInvoke.Shell_NotifyIcon(NOTIFY_ICON_MESSAGE.NIM_SETVERSION, in lpData);
+				}
+				else
+				{
+					// Modify the existing icon
+					PInvoke.Shell_NotifyIcon(NOTIFY_ICON_MESSAGE.NIM_MODIFY, in lpData);
+				}
+			}
+		}
+
+		private void DeleteNotifyIcon()
+		{
+			if (_notifyIconCreated)
+			{
+				_notifyIconCreated = false;
+
+				NOTIFYICONDATAW lpData = default;
+
+				lpData.cbSize = (uint)Marshal.SizeOf(typeof(NOTIFYICONDATAW));
+				lpData.hWnd = _iconWindow.WindowHandle;
+				lpData.guidItem = Id;
+				lpData.uFlags = NOTIFY_ICON_DATA_FLAGS.NIF_MESSAGE | NOTIFY_ICON_DATA_FLAGS.NIF_ICON | NOTIFY_ICON_DATA_FLAGS.NIF_TIP | NOTIFY_ICON_DATA_FLAGS.NIF_GUID | NOTIFY_ICON_DATA_FLAGS.NIF_SHOWTIP;
+
+				// Delete the existing icon
+				PInvoke.Shell_NotifyIcon(NOTIFY_ICON_MESSAGE.NIM_DELETE, in lpData);
+			}
+		}
+
+		private void ShowContextMenu()
+		{
+			PInvoke.GetCursorPos(out var lpPoint);
+
+			DestroyMenuSafeHandle hMenu = PInvoke.CreatePopupMenu_SafeHandle();
+
+			// Generate the classic context menu
+			PInvoke.AppendMenu(hMenu, MENU_ITEM_FLAGS.MF_BYCOMMAND, WM_FILES_CONTEXTMENU_RESTART, "Documentation".GetLocalizedResource());
+			PInvoke.AppendMenu(hMenu, MENU_ITEM_FLAGS.MF_BYCOMMAND, WM_FILES_CONTEXTMENU_QUIT , "SubmitBugReport".GetLocalizedResource());
+			PInvoke.AppendMenu(hMenu, MENU_ITEM_FLAGS.MF_SEPARATOR, 0u, string.Empty);
+			PInvoke.AppendMenu(hMenu, MENU_ITEM_FLAGS.MF_BYCOMMAND, WM_FILES_CONTEXTMENU_DOCSLINK , "Restart".GetLocalizedResource());
+			PInvoke.AppendMenu(hMenu, MENU_ITEM_FLAGS.MF_BYCOMMAND, WM_FILES_CONTEXTMENU_BUGREPORTLINK , "Quit".GetLocalizedResource());
+			PInvoke.SetForegroundWindow(_iconWindow.WindowHandle);
+
+			TRACK_POPUP_MENU_FLAGS tRACK_POPUP_MENU_FLAGS =
+				TRACK_POPUP_MENU_FLAGS.TPM_RETURNCMD |
+				(PInvoke.GetSystemMetricsForDpi((int)SYSTEM_METRICS_INDEX.SM_MENUDROPALIGNMENT, PInvoke.GetDpiForWindow(_iconWindow.WindowHandle)) != 0
+					? TRACK_POPUP_MENU_FLAGS.TPM_RIGHTALIGN
+					: TRACK_POPUP_MENU_FLAGS.TPM_LEFTBUTTON);
+
+			switch (PInvoke.TrackPopupMenuEx(hMenu, (uint)tRACK_POPUP_MENU_FLAGS, lpPoint.x, lpPoint.y, _iconWindow.WindowHandle, null).Value)
+			{
+				case 1:
+					OnDocumentationClicked();
+					break;
+				case 2:
+					OnBugReportClicked();
+					break;
+				case 3:
+					OnRestartClicked();
+					break;
+				case 4:
+					OnQuitClicked();
+					break;
+			}
+		}
+
+		private void OnLeftClicked()
+		{
+		}
+
+		private void OnRestartClicked()
+		{
+			Microsoft.Windows.AppLifecycle.AppInstance.Restart("");
+
+			Program.Pool.Release();
+			Environment.Exit(0);
+		}
+
+		private void OnQuitClicked()
```

Comment:
Will this save the open tabs or is this similar to killing the process from Task Manager?

Reply:
When closed the window, saving is done already.

> this similar to killing the process from Task Manager

Yeah. Just killing the process and threads.
Task Manager, I believe, uses taskkill command or WMI tho.

Reply:
> When closed the window, saving is done already.

That's a problem when resuming on the flip side.

Reply:
> When closed the window, saving is done already.

What if someone uses this option while the window is still open?

Reply:
Wdym?

Reply:
If the window is still open, do the tabs still get saved when using this option?

Reply:
Yes.